Live-Snap/Live-Snap/Controllers/ExportWallpaperViewController.swift

```diff
@@ -58,24 +77,75 @@ class ExportWallpaperViewController: UIViewController, PHLivePhotoViewDelegate {
+            view.addSubview(toolBar)
+            animateToolBarAndLivePhotoView()
+        }
     }
     
     func livePhotoView(_ livePhotoView: PHLivePhotoView, didEndPlaybackWith playbackStyle: PHLivePhotoViewPlaybackStyle) {
-        if !view.subviews.contains(exportButton) {
-            view.addSubview(exportButton)
-            view.isUserInteractionEnabled = true
+        view.isUserInteractionEnabled = true
+        
+        if !livePhotoPreviewFinished {
+            self.view.addSubview(forceTouchNotifierLabel)
+            UIView.animate(withDuration: 0, delay: 1.0, animations: {
+                self.forceTouchNotifierLabel.center = CGPoint(x: self.view.frame.width/2, y: self.view.frame.height/2)
+            })
+            UIView.animate(withDuration: 1.0, delay: 1.0, options: [.autoreverse, .repeat], animations: {
+                self.forceTouchNotifierLabel.transform = CGAffineTransform(scaleX: 1.2, y: 1.2)
+            }, completion: nil)
+            livePhotoPreviewFinished = true
         }
     }
     
+    func animateToolBarAndLivePhotoView() {
+        let IPhoneXSafeAreaBackgroundView = UIView(frame: CGRect(x: 0, y: self.toolBar.frame.origin.y + 45.0, width: self.view.frame.width, height: 34.0))
+        IPhoneXSafeAreaBackgroundView.backgroundColor = UIColor.snapBlack
+
+        UIView.animate(withDuration: 0.5, animations: {
+            if self.view.isIPhoneX() {
+                self.view.frame.size.height -= 34.0
+                self.view.addSubview(IPhoneXSafeAreaBackgroundView)
+            }
+            
+            self.toolBar.frame = CGRect(x: 0, y: self.view.frame.size.height - 45, width: self.view.frame.size.width, height: 45)
+            IPhoneXSafeAreaBackgroundView.frame = CGRect(x: 0, y: self.toolBar.frame.origin.y + self.toolBar.frame.size.height, width: self.view.frame.width, height: 34.0)
+            
+            let orginalLivePhotoViewRect = self.livePhotoPreviewView.frame
+            let newLivePhotoViewRect = CGRect(x: 0, y: UIApplication.shared.statusBarFrame.height, width: self.view.frame.size.width, height: self.view.frame.size.height - (self.toolBar.frame.size.height + UIApplication.shared.statusBarFrame.height))
+            let yOffset = orginalLivePhotoViewRect.midY - newLivePhotoViewRect.midY
+            
+            let orginalTransform = self.livePhotoPreviewView.transform
+            let transformScale = newLivePhotoViewRect.size.height / orginalLivePhotoViewRect.size.height
+            let scaledTransform = orginalTransform.scaledBy(x: transformScale, y: transformScale)
+            let scaledTransformAndTranslate = scaledTransform.translatedBy(x: 0.0, y: -yOffset)
+            self.livePhotoPreviewView.transform = scaledTransformAndTranslate
+        })
+    }
+    
+    @objc func backButtonWasPressed() {
+        let cropWallpaperViewController = CropWallpaperViewController()
+        cropWallpaperViewController.imageToCrop = System.shared.imageToCrop!
```

Comment:
system should not have to keep track of imageToCrop as that is not something global